### PR TITLE
object_horse_ganon, object_horse_link_child, object_horse_normal and object_horse_zelda OK

### PR DESCRIPTION
**Pull request by AngheloAlf**
:racehorse:

### CODE REVIEW

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 1]
```diff
@@ -32,7 +33,8 @@ const ActorInit En_Horse_Link_Child_InitVars = {
     (ActorFunc)EnHorseLinkChild_Draw,
 };
 
-static AnimationHeader* sAnimations[] = { 0x060043E4, 0x06004B08, 0x060053F0, 0x0600360C, 0x06002F98 };
+static AnimationHeader* sAnimations[] = { &gChildEponaIdleAnim, &gChildEponaWhinnyAnim, &gChildEponaWalkingAnim,
+                                          &gChildEponaTrottingAnim, &gChildEponaGallopingAnim };
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:37]: can you add a trailing comma

File: src/overlays/actors/ovl_En_Viewer/z_en_viewer.h [hunk 2]
```diff
@@ -18,7 +18,11 @@ typedef struct {
     /* 0x06 */ u8 actorShadowDrawFuncIndex;
     /* 0x07 */ u8 unk_7;
     /* 0x08 */ u8 drawFuncIndex;
-    /* 0x0C */ FlexSkeletonHeader* unk_C;
+    /* 0x0C */ union {
+                    void* ptr; // prevents compiler warnings
+                    SkeletonHeader* normal;
+                    FlexSkeletonHeader* flex;
+                } unk_C;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Viewer/z_en_viewer.h:25]: Does the code in this actor use any fields from `SkeletonHeader` or `FlexSkeletonHeader`? If not, we can probably just treat pointers to skeleton headers as `void*` everywhere, and avoid any union or casts.

Comment (AngheloAlf) [reply src/overlays/actors/ovl_En_Viewer/z_en_viewer.h]: Yep, that works.

**Comment by Roman971** [comment]
I think you can also make the skeleton header pointers `void*` in the functions. This way you wouldn't need any casts inside them.

**Comment by fig02** [comment]
small undefined syms conflict to resolve

File: src/overlays/actors/ovl_En_Viewer/z_en_viewer.c [hunk 3]
```diff
@@ -178,14 +164,15 @@ void EnViewer_Destroy(Actor* thisx, GlobalContext* globalCtx) {
...
 }
 
-void func_80B2A300(EnViewer* this, GlobalContext* globalCtx, FlexSkeletonHeader* skeletonHeaderSeg,
+void func_80B2A300(EnViewer* this, GlobalContext* globalCtx, void* skeletonHeaderSeg,
                    AnimationHeader* animationSeg) {
     s16 params = this->actor.params >> 8;
 
     if (params == 2 || params == 3 || params == 5 || params == 7 || params == 8 || params == 9) {
-        SkelAnime_InitFlex(globalCtx, &this->skin.skelAnime, skeletonHeaderSeg, NULL, NULL, NULL, 0);
+        SkelAnime_InitFlex(globalCtx, &this->skin.skelAnime, (FlexSkeletonHeader*)skeletonHeaderSeg, NULL, NULL, NULL,
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Viewer/z_en_viewer.c:172]:
```suggestion
        SkelAnime_InitFlex(globalCtx, &this->skin.skelAnime, skeletonHeaderSeg, NULL, NULL, NULL,
```

File: src/overlays/actors/ovl_En_Viewer/z_en_viewer.c [hunk 4]
```diff
@@ -196,18 +183,18 @@ void func_80B2A300(EnViewer* this, GlobalContext* globalCtx, FlexSkeletonHeader*
     }
 }
 
-void func_80B2A448(EnViewer* this, GlobalContext* globalCtx, FlexSkeletonHeader* skeletonHeaderSeg,
+void func_80B2A448(EnViewer* this, GlobalContext* globalCtx, void* skeletonHeaderSeg,
                    AnimationHeader* animationSeg) {
-    SkelAnime_InitFlex(globalCtx, &this->skin.skelAnime, skeletonHeaderSeg, NULL, NULL, NULL, 0);
+    SkelAnime_InitFlex(globalCtx, &this->skin.skelAnime, (FlexSkeletonHeader*)skeletonHeaderSeg, NULL, NULL, NULL, 0);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Viewer/z_en_viewer.c:188]:
```suggestion
    SkelAnime_InitFlex(globalCtx, &this->skin.skelAnime, skeletonHeaderSeg, NULL, NULL, NULL, 0);
```

**Review by Roman971 — COMMENTED** [review]
Last small things but otherwise this looks good to me.

Also I suggest running the formatter again after all this, just in case.

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**